### ISSUE TITLE
Inconsistent audio samples between steps in training

### ISSUE THREAD

**Issue by faceair** [fishaudio/fish-speech#17]
<img width="783" alt="image" src="https://github.com/fishaudio/fish-speech/assets/3659110/d924e8a9-77e9-457c-8698-89278a0a40d8">

I am attempting to finetune the `VQGAN`. On the audio page, under sample-0/wavs/prediction, the sound played at step 0 is a inferred voice segment (which is correct). However, at step 2000, the sound played is a different voice clip from the training set, and the content of these two segments is different.

I'm unsure where the issue lies and how to troubleshoot it.

**Comment by leng-yue** [comment]
Sometime this is caused by lagging of tensorboard. Can you try restart tensorboard?

**Comment by faceair** [comment]
I've also noticed the gt dataset has steps. In my sample-0/wavs/gt, step 0 and step 2000 are not the same sample.

The current situation appears as follows:
|                | sample-0/wavs/gt  | sample-0/wavs/prediction |
|----------------|-------------------|--------------------------|
| step 0         | raw_sample_1      | inferred_voice_1         |
| step 2000      | raw_sample_2      | raw_sample_2             |
| step 4000      | raw_sample_2      | raw_sample_2             |
| step 6000      | raw_sample_2      | raw_sample_2             |

**Comment by leng-yue** [comment]
I double checked the code and can't find any issue. Can you check the checksum of `gt` and `prediction` at step 2000/4000?

**Comment by faceair** [comment]
<img width="518" alt="image" src="https://github.com/fishaudio/fish-speech/assets/3659110/94288369-bf65-44b3-9834-4fb39f298d7e">

It appears that the only issue is that the gt data was replaced between step 0 and step 2000.

The prediction validation set is generated based on gt, and in my dataset, it seems that due to excessive similarity, I failed to correctly distinguish whether the audio was generated or not; it's too perfect.

**Comment by faceair** [comment]
I have two more questions:

1. What should my input be when validating after finetuning? Should it be the voice of the same person? To see if the output voice perfectly imitates this voice?
2. During the final inference, how should I fill in the prompt-text? What can I control, and can you provide some more specific examples?

**Comment by leng-yue** [comment]
That makes sense, the first validating in lightning is sanity check, it didn't cover all samples.

1. It should be the same person, but just make sure your training and validation set is not overlaping.
2. You should fill the prompt-text in the semantic (llama) model, instead of vqgan.
